duckduckgo_search/cli.py

```diff
@@ -203,12 +223,16 @@ def text(keywords, region, safesearch, timelimit, backend, output, download, thr
         max_results=max_results,
     )
     keywords = _sanitize_keywords(keywords)
-    filename = f"text_{keywords}_{datetime.now():%Y%m%d_%H%M%S}"
+    if output.endswith(".json") or output.endswith(".csv"):
+        filename = output.rsplit(".", 1)[0]
+    else:
+        filename = f"text_{keywords}_{datetime.now():%Y%m%d_%H%M%S}"
+
     if output == "print" and not download:
         _print_data(data)
-    elif output == "csv":
+    elif output == "csv" or output.endswith(".csv"):
         _save_csv(f"{filename}.csv", data)
-    elif output == "json":
+    elif output == "json" or output.endswith(".json"):
         _save_json(f"{filename}.json", data)
     if download:
         _download_results(keywords, data, proxy=proxy, threads=threads, verify=verify)
```

Comment:
if `output` is dir and `download` is not None, pass `output` to def `_download_results` and save results to `output` dir